addons/l10n_pe_website_sale/static/src/js/website_sale.js

```diff
@@ -0,0 +1,104 @@
+/** @odoo-module **/
+import { WebsiteSale } from "@website_sale/js/website_sale";
+
+WebsiteSale.include({
+    events: Object.assign({}, WebsiteSale.prototype.events, {
+        "change select[name='state_id']": "_onChangeState",
+        "change select[name='city_id']": "_onChangeCity",
+    }),
+    start: function () {
+        this.selectCities = document.querySelector("select[name='city_id']");
+        this.selectDistricts = document.querySelector("select[name='l10n_pe_district']");
+        this.cityBlock = document.querySelector(".div_city");
+        this.autoFormat = document.querySelector(".checkout_autoformat");
+        this.selectState = document.querySelector("select[name='state_id']");
+        return this._super.apply(this, arguments);
+    },
+    _changeState: function () {
+        if (!this.selectState.value) {
+            return;
+        }
+        return this._rpc({
+            route: "/shop/state_infos/" + this.selectState.value,
+        }).then(data => {
+            // populate cities and display
+            if (data.cities?.length) {
+                this.selectCities.innerHTML = "";
+                data.cities.forEach(city => {
+                    let opt = document.createElement("option");
+                    opt.textContent = city[1];
+                    opt.value = city[0];
+                    opt.setAttribute("data-code", city[2]);
+                    this.selectCities.appendChild(opt);
+                });
+                this.selectCities.parentElement.style.display = "block";
+            } else {
+                this.selectCities.value = "";
+                this.selectCities.parentElement.style.display = "none";
+            }
+            this.selectDistricts.value = "";
+            this.selectDistricts.parentElement.style.display = "none";
+        });
+    },
+    _changeCity: function () {
+        if (!document.getElementById("city_id").value) {
+            return;
+        }
+        this._rpc({
+            route: "/shop/city_infos/" + document.getElementById("city_id").value,
+        }).then(data => {
+            // populate districts and display
+            if (data.districts.length) {
+                this.selectDistricts.innerHTML = "";
+                data.districts.forEach(district => {
+                    let opt = document.createElement("option");
+                    opt.textContent = district[1];
+                    opt.value = district[0];
+                    opt.setAttribute("data-code", district[2]);
+                    this.selectDistricts.appendChild(opt);
+                });
+                this.selectDistricts.parentElement.style.display = "block";
+            } else {
+                this.selectDistricts.value = "";
+                this.selectDistricts.parentElement.style.display = "none";
+            }
+        });
+    },
+    _onChangeState: function (ev) {
```

Comment:
```suggestion
    _onChangeState: function () {
```
No need for the argument if you don't use it.

Reply:
Done